dotnet-authserver/src/TeacherIdentity.AuthServer/Controllers/AuthorizationController.cs

```diff
@@ -126,13 +114,16 @@ public async Task<IActionResult> Authorize()
                 userRequirements,
                 GetCallbackUrl(journeyId),
                 startedAt: _clock.UtcNow,
-                authenticateResult.Succeeded != true,
-                authStateInitData,
                 sessionId,
                 oAuthState: new OAuthAuthorizationState(request.ClientId!, request.Scope!, request.RedirectUri)
                 {
                     TrnRequirementType = trnRequirementType
-                });
+                },
+                authenticateResult.Succeeded != true);
+
+            var signedInUser = await GetSignedInUser(authenticateResult, userRequirements);
+            var existingUserClaimsPrincipal = await InitializeAuthenticationState(signedInUser, trnToken, authenticationState);
```

Comment:
This is a bit of an awkward way to retrieve the claims principal for the new signed in user, perhaps there's a better way of doing it?

Reply:
Seems reasonable to me and I can't think of anything better.